group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2603,9 +2648,54 @@ public static Map<String, ResolvedRegularExpression> refreshRegularExpressions(
         return result;
     }
 
+    /**
+     * This method filters the topics in the resolved regexes
+     * that the member is authorized to describe.
+     *
+     * @param context           The request context.
+     * @param authorizer        The authorizer.
+     * @param resolvedRegexes   The map of the regex patter and its set of matched topics.
+     * @return The set of topics that the member is not authorized to describe.
+     */
+    private static Set<String> filterTopicDescribeAuthorizedTopics(
+        RequestContext context,
+        Optional<Authorizer> authorizer,
+        Map<String, Set<String>> resolvedRegexes
+    ) {
+        if (authorizer.isPresent()) {
+            Map<String, Integer> topicNameCount = new HashMap<>();
+            resolvedRegexes.values().forEach(topicNames ->
+                topicNames.forEach(topicName ->
+                    topicNameCount.compute(topicName, Utils::incValue)
+                )
+            );
+
+            List<Action> actions = topicNameCount.entrySet().stream().map(entry -> {
+                ResourcePattern resource = new ResourcePattern(TOPIC, entry.getKey(), LITERAL);
+                return new Action(DESCRIBE, resource, entry.getValue(), true, true);
+            }).collect(Collectors.toList());
+
+            List<AuthorizationResult> authorizationResults = authorizer.get().authorize(context, actions);
+            Set<String> deniedTopics = new HashSet<>();
+            IntStream.range(0, actions.size()).forEach(i -> {
+                if (authorizationResults.get(i) == AuthorizationResult.DENIED) {
+                    String deniedTopic = actions.get(i).resourcePattern().name();
+                    deniedTopics.add(deniedTopic);
+                }
+            });
+
+            resolvedRegexes.forEach((__, topicNames) -> topicNames.removeIf(deniedTopics::contains));
+
+            return deniedTopics;
+        }
+        return Collections.emptySet();
+    }
+
+
     /**
      * Handle the result of the asynchronous tasks which resolves the regular expressions.
      *
+     * @param groupId                       The group id.
```

Comment:
Could we pass the member id and extend the log `"[GroupId {}] Received updated regular expressions: {}.` to include it? This is useful to know that they were resolved based on the context of the member X.